core/src/main/java/hudson/model/Node.java

```diff
@@ -115,6 +118,18 @@ public boolean isHoldOffLaunchUntilSave() {
     }
 
     /**
+     * {@inheritDoc}
+     * @since 1.635
+     */
+    @Override
+    public void save() throws IOException {
+        final Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            jenkins.updateNode(this);
+        }
```

Comment:
surely this should throw an exception if Jenkins is null?
otherwise the caller will think that the `save` operation has succeeded.